coordinator/src/main/scala/filodb.coordinator/queryplanner/DefaultPlanner.scala

```diff
@@ -338,17 +342,24 @@ trait  DefaultPlanner {
                                         sqww: SubqueryWithWindowing
                                       ) : PlanResult = {
     // absent over time is essentially sum(last(series)) sent through AbsentFunctionMapper
-    innerExecPlan.plans.foreach(
-      _.addRangeVectorTransformer(PeriodicSamplesMapper(
-        sqww.startMs, sqww.stepMs, sqww.endMs,
-        window,
-        Some(InternalRangeFunction.lpToInternalFunc(RangeFunctionId.Last)),
-        qContext,
-        stepMultipleNotationUsed = false,
-        Seq(),
-        offsetMs,
-        rawSource = false
-      ))
+     val realStartMs = sqww.atMs.getOrElse(sqww.startMs)
+     val realEndMs = sqww.atMs.getOrElse(sqww.endMs)
+     val realStep = sqww.atMs.map(_ => 0L).getOrElse(sqww.stepMs)
+
+     innerExecPlan.plans.foreach(plan => {
+       plan.addRangeVectorTransformer(PeriodicSamplesMapper(
+         realStartMs, realStep, realEndMs,
```

Comment:
Should the periodic mapper be passed the `real` time parameters or the actual ones? It gives the correct timestamps when the `real` parameters are passed?